enhancements/cluster-logging/forward_to_cloudwatch.md

```diff
@@ -0,0 +1,226 @@
+---
+title: forward_to_cloudwatch
+authors:
+  - "@alanconway"
+reviewers:
+  - "@jcantrill"
+  - "@jeremyeder"
+approvers:
+creation-date: 2020-12-17
+last-updated: 2020-12-17
+status: implementable
+see-also:
+superseded-by:
+---
+
+# Forward to Cloudwatch
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+[Amazon Cloudwatch][aws-cw] is a hosted monitoring and log storage service.
+This proposal extends the `ClusterLogForwarder` API with an output type for Cloudwatch.
+
+## Motivation
+
+Amazon Cloudwatch is a popular log store.
+We have requests from external and  Red Hat-internal customers to support it.
+
+### Goals
+
+Enable log forwarding to Cloudwatch.
+
+### Non-Goals
+
+Enable Cloudwatch metric collection.
+
+## Proposal
+
+### Cloudwatch streams and groups
+
+[Cloudwatch][concepts] defines *log groups* and *log streams*. To paraphrase the documentation:
+
+>  A log stream is a sequence of log events that share the same source ... For example, an Apache access log on a specific host.
+
+>  Log groups define groups of log streams that share the same retention, monitoring, and access control settings ... For example, if you have a separate log stream for the Apache access logs from each host, you could group those log streams into a single log group called MyWebsite.com/Apache/access_log.
+
+In other words a *log stream* corresponds to the smallest log-producing units.
+A *log group* is a collection of related *log streams*.
+
+We must consider *container logs*, and *node logs*.
+- "application" logs are always container logs
+- "infrastructure" logs are a mix of container and node logs
+- "audit" logs are all node logs.
+
+For *container-scoped* logs we auto-create a log stream for *each container*, using the container UUID as the log stream name.
```

Comment:
The POC is using the fluent tag which is the full file path on the node.  Do we think its necessary to further burden the collector to change it to something else?

Reply:
Why send the entire fluent tag?  Isn't the UUID much shorter?

Reply:
The question is, can the user find out what is the full path of the container log file through some Kube API?  We know they can retrieve the UUID of the container.

Reply:
Perhaps a rewording to drop "scoped" which hasnt' been introduced before this?  `For *container-logs*, we auto-create a log stream for *each container* using its UUID as the log stream name.`

Also, it is not clear the log forwarder chooses to auto-create, it seems the CloudWatch instance has to be configured to allow such auto-creation.  Might be worth mentioning as a requirement.